app/api/handler/account/cookie.go

```diff
@@ -48,16 +51,21 @@ func deleteTokenCookieIfPresent(r *http.Request, w http.ResponseWriter, cookieNa
 	cookie.Value = ""
 	cookie.Expires = time.UnixMilli(0) // this effectively tells the browser to delete the cookie
 
+	// Set the Secure attribute to true
+	cookie.Secure = true
+	
 	http.SetCookie(w, cookie)
 }
 
 func newEmptyTokenCookie(r *http.Request, cookieName string) *http.Cookie {
+	isSecure := r.TLS != nil // Check if the request is using HTTPS
```

Comment:
Not sure how `r.TLS` is fixing anything specific, could you elaborate?
One scenario I'm thinking of improving is gitness behind a reverse proxy with TLS offloading - in that case `r.TLS` would be `nil` and `r.URL.Proto` would be http.

So an improvement would be to check for the availability of a `X-Forwarded-Proto` header (otherwise fall back to `r.URL.Proto` or `r.TLS`) to determine whether `cookie.Secure` should be set?

cc @enver-bisevac

Reply:
I agree @johannesHarness